client/js/commands/join.js

```diff
@@ -8,14 +8,37 @@ exports.input = function(args) {
 	const {vueApp} = require("../vue");
 
 	if (args.length > 0) {
-		const channel = args[0];
+		let channel = args[0];
 
 		if (channel.length > 0) {
+			const chanTypes = vueApp.activeChannel.network.serverOptions.CHANTYPES;
+			let hasPrefix = false;
+
+			for (const prefix of chanTypes) {
+				if (channel.startsWith(prefix)) {
+					hasPrefix = true;
+					break;
+				}
+			}
+
+			if (!hasPrefix && chanTypes && chanTypes.length > 0) {
```

Comment:
You check `chanTypes` here but already tried to loop above. both of them need to be wrapped.